salt/states/zabbix_host.py

```diff
@@ -182,40 +210,35 @@ def _interface_format(interfaces_data):
     # Get and validate proxyid
     proxy_hostid = "0"
     if "proxy_host" in kwargs:
+        proxy_host = kwargs.pop("proxy_host")
         # Test if proxy_host given as name
-        if isinstance(kwargs["proxy_host"], six.string_types):
+        if isinstance(proxy_host, str):
             try:
                 proxy_hostid = __salt__["zabbix.run_query"](
                     "proxy.get",
                     {
                         "output": "proxyid",
                         "selectInterface": "extend",
-                        "filter": {"host": "{0}".format(kwargs["proxy_host"])},
+                        "filter": {"host": "{}".format(proxy_host)},
                     },
                     **connection_args
                 )[0]["proxyid"]
             except TypeError:
-                ret["comment"] = "Invalid proxy_host {0}".format(kwargs["proxy_host"])
+                ret["comment"] = "Invalid proxy_host {}".format(proxy_host)
                 return ret
         # Otherwise lookup proxy_host as proxyid
         else:
             try:
                 proxy_hostid = __salt__["zabbix.run_query"](
                     "proxy.get",
-                    {
-                        "proxyids": "{0}".format(kwargs["proxy_host"]),
-                        "output": "proxyid",
-                    },
+                    {"proxyids": "{}".format(proxy_host), "output": "proxyid"},
                     **connection_args
                 )[0]["proxyid"]
             except TypeError:
-                ret["comment"] = "Invalid proxy_host {0}".format(kwargs["proxy_host"])
+                ret["comment"] = "Invalid proxy_host {}".format(proxy_host)
                 return ret
 
-    if "inventory" not in kwargs:
-        inventory = {}
-    else:
-        inventory = kwargs["inventory"]
+    inventory = kwargs.pop("inventory", None)
```

Comment:
I think in this case it would need to be `kwargs.pop("inventory", {})`

Reply:
In the original code there is a safeguard if `inventory` is passed as `None`. Don't know the use case for this, probably formulas that always pass the `inventory` argument but with `None` when it is not needed. I try to not change the existent parsing.

If the user pass the `inventory = None`, the `kwargs.pop("inventory", {})` will pass the same `None` to the next steps and the `new_inventory` population will fail. That´s why I use `kwargs.pop("inventory", None)` and kept the conversion from `None` to `{}` in next line.

Reply:
Sounds good.